pyomo/util/calc_var_value.py

```diff
@@ -190,7 +195,13 @@ def calculate_variable_from_constraint(variable, constraint,
                 "expression.\n\tPlease provide a different initial guess "
                 "or enable the linesearch if you have not.")
             raise
-        fpk = value(expr_deriv)
+
+        if diff_mode in {differentiate.Modes.sympy, differentiate.Modes.reverse_symbolic}:
```

Comment:
Note that this test is somewhat expensive, (you have to create the set and then check for membership in it).  I think I would rather do the test once above (line 162) and if it is False, then set `expr_deriv = None`.  Then here we can just do:
```python
if expr_deriv is None:
    fpk = differentiate(expr, wrt=variable, mode=diff_mode)
else:
    fpk = value(expr_deriv)
```

Reply:
Done!